### PR TITLE
Analytics Hub:  Improve date range selector UI

### PR DESCRIPTION
**Pull request by ThomazFB**
Changelog
==========
Reorganize the Analytics Hub selector UI to match the iOS version and rename the internal UI naming pattern for better identification with the range Selection class.

Screenshots
==========
| Before  | After |
| ------------- | ------------- |
| Screenshot_20230110_001942 | Screenshot_20230110_001750 |

Disclaimer
==========
Since it will take a bunch of changes to achieve the complete scope, this PR is targeted to a feature branch called `improve/analytics-hub` that will aggregate everything, this way we can review piece by piece while keeping the production version intact.

How to Test
==========
1. Build the app and open the Analytics Hub. If possible, keep a second device/emulator running the Hub from the `trunk` branch
2. Verify that Hub still opens following the selected tab from the My Store view. E. g. `This Year` tab should open the Hub selected in the `Year to Date` range.
3. Verify that all options for date range still exist and are the same from the original Hub

Update release notes:

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/analytics/daterangeselector/AnalyticsDateRangeCardView.kt [hunk 1]
```diff
@@ -18,11 +21,19 @@ class AnalyticsDateRangeCardView @JvmOverloads constructor(
...
     }
 
-    fun updateToText(toDatePeriod: String) {
-        binding.tvToDate.text = toDatePeriod
+    fun updatePreviousRange(previousRange: String) {
+        SpannableStringBuilder()
+            .append(resources.getString(R.string.date_compared_to))
+            .append(" ")
+            .bold { append(previousRange) }
+            .let { binding.previousRangeDescription.text = it }
```
Comment (atorresveiga) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/analytics/daterangeselector/AnalyticsDateRangeCardView.kt:33]: Nice!

**Review by wzieba — APPROVED** [review]
Looks good, I like the new design much more!

**Comment by atorresveiga** [comment]
Great work @ThomazFB, the code looks good! I also liked more the new design.

I want to mention that I found that products and visitor stats have different values. Not sure if this is expected. 🤔

**Comment by ThomazFB** [comment]
Thanks for testing those numbers @atorresveiga! 🙇  I detected this issue with the products and visitors data too, and the fix will soon come through [this PR](https://github.com/woocommerce/woocommerce-android/pull/8168)! 

PS: I recommend using the My Store product numbers as the truth source for comparison.

**Pull request merged**